rattlesnake/01-a-class-of-students/soln-python-restful/main.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+
+import re,sys
+
+# The "database"
+data = {}
+# The "session"
+session = {}
+
+# Internal functions of the "server"-side application
+def error_state():
+    return "Something wrong", ["get", "default", None]
+
+# The "server"-side application handlers
+def default_get_handler(args):
+    rep = "What would you like to do?"
+    rep += "\n1 - Quit" + "\n2 - Upload File"
+    links = {"1" : ["post", "execution", None], "2" : ["get", "file_form", None]}
+    return rep, links
+
+def quit_handler(args):
+    for filename in data:
+        with open(filename,"w") as f:
+            for row in data[filename]:
+                f.write(','.join(row))
+                f.write("\n")
+
+    sys.exit("Goodbye m'lord!")
+
+def add_get_handler(args):
+    rep = "Enter Name, Phone, Marks1, Marks2, Marks3, Marks4: "
+    links = ["post", "add"]
+    return rep, links
+
+def add_post_handler(args):
+    def add_to_data(new_data):
+        for key in data:
+            data[key].append(new_data[0].split(','))
+
+    if args == None:
+        return error_state()
+    new_data = args
+    try:
+        add_to_data(new_data)
+    except:
```

Comment:
As a rule of thumb, add exception type if you know what exception might be raised. For e.g. here I can see KeyError and IndexError (and Possibly AttributeError).